### PR TITLE
HTTP Exception Handling

### PR DESCRIPTION
**Pull request by vilyapilya**
If logger receives an unsuccessful response, it will try to reconnect with the server 3 times. If it still fails, it will save the buffer on the logger instance and when the next time when a new message is pushed into buffer, it will merge the current buffer with the unsend content of the previous buffer. 
When it merges the unsent info with the current buffer, it cleans the temp holder for the unsent info. 
The holder for the unsent info has a limit to grow

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -227,6 +235,31 @@ Logger.prototype._bufferLog = function(message) {
     }
 };
 
+Logger.prototype._sendRequest = function(_config, dbgInfoAndAttemps, callback) {
+    axios(_config)
+        .then(function(response) {
+            if (response && response.status === 200) {
+                return {
```
Comment (smusali) [inline lib/logger.js]: Here it doesn't return `callback` but in `L248`, it returns `callback` - so, the question is which one is correct - shouldn't this always return `callback`?

**Review by smusali — CHANGES_REQUESTED** [review]
Have you seen how it's handled in [CloudWatch](https://github.com/logdna/aws-cloudwatch/blob/master/node.js/index.js)? Is there any similarity in handling or this one should be handled differently than the CloudWatch one?

File: lib/logger.js (hunk 1 above)
Comment (vilyapilya) [reply lib/logger.js]: oh good catch. Thank you!

Comment (vilyapilya) [reply lib/logger.js]: To your first comment: I see that CloudWatch retries to connect. I also retry to connect if it fails but I also save the failed lines in memory so they don't get lost.

**Comment by smusali** [comment]
@vilyapilya, there are merge conflicts due to the previous PR #60 merge. Can you fix/resolve them? Thanks in advance!

**Pull request closed without merging**